convene-web/app/models/demo_workspace.rb

```diff
@@ -0,0 +1,47 @@
+class DemoWorkspace
+  # Creates the demo workspace, but only on environments which are
+  # configured to include the demo workspace.
+  def self.prepare
+    return unless Feature.enabled?(:demo)
+
+    Factory.new(Client).find_or_create_workspace!
+  end
+
+  # Assembles the DemoWorkspace using the provided Client and Workspace
+  # classes.
+  class Factory
+    attr_accessor :client_repository
+
+    # These are the Rooms we expect the DemoWorkspace to have by default.
+    # Our customer team leverages them for their demonstrations of Convene's
+    # feature set.
+    DEMO_ROOMS = ["Zee's Desk", "Vivek's Desk", 'Water Cooler', 'The Ada Lovelace Room'].freeze
+
+    # @param [ActiveRecord::Relation<Client>] client_repository Where to ensure there
+    #  is a Zinc Client with the Convene Demo workspace
+    def initialize(client_repository)
+      self.client_repository = client_repository
```

Comment:
I missed this during review, I don't see any code referencing `client_repository`, did I miss anything?

Reply:
Woops, it should be in 44.

Reply:
Got it, makes sense.

Reply:
Erm, I am thinking why do we want to leave this interface open since code in line 44 needs to be `Client` anyway?

Reply:
So, I have a habit (which may or may not be useful) of not referencing model classes whenever I can help it. Instead, I treat the Model class like a [Repository](https://8thlight.com/blog/mike-ebert/2013/03/23/the-repository-pattern.html). 

It's not particularly "necessary" by any means, but I have grown to appreciate explicit seams like this.  You can see the complete implementation here: https://github.com/zinc-collective/convene/pull/67/files#diff-9740bfa8740b2c2a549405d21a90605a

Reply:
Moved to it's own PR so that 67 can stay focused - https://github.com/zinc-collective/convene/pull/68